src/metabase/events/view_log.clj

```diff
@@ -57,6 +57,17 @@
       (catch Throwable e
         (log/warnf e "Failed to process view event. %s" topic)))))
 
+(derive ::dashboard-queried :metabase/event)
+(derive :event/dashboard-queried ::dashboard-queried)
+
+(m/defmethod events/publish-event! ::dashboard-queried
+  "Handle processing for a dashboard query being run"
+  [topic {:keys [object-id] :as _event}]
+  (try
+    (t2/update! :model/Dashboard {:id object-id} {:last_viewed_at :%now})
```

Comment:
The dashboard model has an `updated_at` column that is automatically updated by toucan with this hook. 
https://github.com/metabase/metabase/blob/2d7891b8ed9f4dc29d12afe37c666d625bf686fa/src/metabase/models/interface.clj#L440-L443

I'm not sure what this column is used for, but its meaning will change substantially if we update it every time the dashboard is viewed. I've avoided this issue previously by skipping model hooks with this trick:
https://github.com/metabase/metabase/blob/2d7891b8ed9f4dc29d12afe37c666d625bf686fa/src/metabase/events/view_log.clj#L21-L24

Maybe `updated_at` is not actually used for anything, so this is okay. But I think we should avoid using toucan hooks here for performance reasons anyway. It will trigger extra queries we can easily avoid here.